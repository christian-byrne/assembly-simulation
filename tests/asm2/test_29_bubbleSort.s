.data

.globl	printInts
.globl	printWords
.globl	bubbleSort

.globl	printInts_howToFindLen
.globl	intsArray
.globl	intsArray_END
.globl	intsArray_len
.globl	theString

printInts:
	.byte	0
printWords:
	.byte	0
bubbleSort:
	.byte	1

printInts_howToFindLen:
	.half	0	# 0 - read intsArray_len; 1 - calc intsArray_END-intsArray; 2 - null terminated

intsArray:
	.word 3
	.word 4
	.word 2
	.word 2
	.word 5
	.word 0

intsArray_END:

intsArray_len:
	.word	5

theString:
	.asciiz	"wgwa awy awy"



# ----------- main() -----------
.text

.globl	main

main:
	jal studentMain
	

