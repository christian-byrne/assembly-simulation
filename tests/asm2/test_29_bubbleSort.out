Swap at: 1
Swap at: 2
Swap at: 0
Swap at: 1
